### PR TITLE
SONAR-7630 Do not use MySQL TINYINT(1) for non-boolean columns

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: sonar-application/src/main/assembly/conf/sonar.properties [hunk 1]
```diff
@@ -17,9 +17,7 @@
 #----- Embedded Database (default)
 # H2 embedded database server listening port, defaults to 9092
 #sonar.embeddedDatabase.port=9092
-
-
-#----- MySQL 5.x
+#----- MySQL 5.6 or greater
```
Comment (sns-seb) [inline sonar-application/src/main/assembly/conf/sonar.properties]: you could associate this commit to SONAR-7260 (ie. a recent JIRA I found related to min version of supported DBs)

File: server/sonar-server/src/main/java/org/sonar/server/platform/platformlevel/PlatformLevel2.java [hunk 2]
```diff
@@ -72,5 +73,7 @@ protected void configureLevel() {
       // DB migration
       PlatformDatabaseMigrationExecutorServiceImpl.class,
       PlatformDatabaseMigration.class);
+
+    add(MigrationStepModule.class);
```
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/platform/platformlevel/PlatformLevel2.java]: why a separate add ?

File: sonar-db/src/test/java/org/sonar/db/version/v56/FixTypeOfRuleTypeOnMysqlTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    underTest.execute(context);
+
+    verify(context).execute("ALTER TABLE rules MODIFY COLUMN rule_type TINYINT (2)");
+    verify(context).execute("ALTER TABLE issues MODIFY COLUMN issue_type TINYINT (2)");
+  }
+
+  @Test
+  public void do_not_alter_columns_if_fresh_mysql_install() throws SQLException {
+    prepare(-1, new MySql());
```
Comment (sns-seb) [inline sonar-db/src/test/java/org/sonar/db/version/v56/FixTypeOfRuleTypeOnMysqlTest.java]: nice usage of constants in other methods, why not here too?

File: server/sonar-server/src/main/java/org/sonar/server/platform/platformlevel/PlatformLevel2.java (hunk 2 above)
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/platform/platformlevel/PlatformLevel2.java]: copy past from PlatformLevel1

File: sonar-db/src/test/java/org/sonar/db/version/v56/FixTypeOfRuleTypeOnMysqlTest.java (hunk 3 above)
Comment (simonbrandhof) [inline sonar-db/src/test/java/org/sonar/db/version/v56/FixTypeOfRuleTypeOnMysqlTest.java]: indeed, I simply forgot.

**Comment by simonbrandhof** [comment]
feedback applied.

**Comment by sns-seb** [comment]
LGTM

**Pull request closed without merging**